### PR TITLE
Feat/ss 167

### PR DESCRIPTION
**Pull request by spiropulo**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,6 +3,11 @@ __pycache__/
 *.py[cod]
 *$py.class
 
+.project
+.pydevproject
+package-lock.json
```
Comment (sidler-mozilla) [inline .gitignore]: do not ignore this file

File: subhub/api/webhooks/stripeChargeSucceededProcessor.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
+from subhub.api.webhooks.abstractStripeWebhookProcessor import AbstractStripeWebhookProcessor
```
Comment (sidler-mozilla) [inline subhub/api/webhooks/stripeChargeSucceededProcessor.py]: remove stubbed code that we are not using

File: subhub/featuretests/stripe_webhooks/test_stripe_webhook_calls.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        runTest("badpayload.json")
+    except ValueError as e:
+        assert "this.will.break is not supported" == str(e)
+
+
+def runTest(fileName):
+    __location__ = os.path.realpath(os.path.join(os.getcwd(), os.path.dirname(__file__)))
+    f = open(os.path.join(__location__, fileName));
+
+    with f:
```
Comment (sidler-mozilla) [inline subhub/featuretests/stripe_webhooks/test_stripe_webhook_calls.py]: why not ```with open(blah``` ?

File: subhub/api/webhooks/stripeChargeSucceededProcessor.py (hunk 2 above)
Comment (spiropulo) [reply subhub/api/webhooks/stripeChargeSucceededProcessor.py]: what stub code?

File: subhub/featuretests/stripe_webhooks/test_stripe_webhook_calls.py (hunk 3 above)
Comment (spiropulo) [reply subhub/featuretests/stripe_webhooks/test_stripe_webhook_calls.py]: I have no idea

File: .gitignore (hunk 1 above)
Comment (spiropulo) [reply .gitignore]: OK

File: subhub/featuretests/stripe_webhooks/test_stripe_webhook_calls.py (hunk 3 above)
Comment (sidler-mozilla) [reply subhub/featuretests/stripe_webhooks/test_stripe_webhook_calls.py]: http://cmdlinetips.com/2016/01/opening-a-file-in-python-using-with-statement/

File: subhub/api/webhooks/stripeChargeSucceededProcessor.py (hunk 2 above)
Comment (sidler-mozilla) [reply subhub/api/webhooks/stripeChargeSucceededProcessor.py]: https://github.com/mozilla/subhub/blob/feat/SS-167/subhub/api/webhooks/stripeChargeDisputeFundsWithdrawnProcessor.py

You have several files that do nothing other than print in the run function.  YAGNI: you aint gonna need it; remove these

also consider consolidating your class definitions into one file, especially if they are only going to be several lines each.

File: subhub/api/webhooks/stripeWebhookPipeline.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
+from subhub.api.webhooks.stripeChargeCapturedProcessor import StripeChargeCapturedProcessor
+from subhub.api.webhooks.stripeChargeDisputeClosedProcessor import StripeChargeDisputeClosedProcessor
+from subhub.api.webhooks.stripeChargeDisputeCreatedProcessor import StripeChargeDisputeCreatedProcessor
+from subhub.api.webhooks.stripeChargeDisputeFundsReinstatedProcessor import StripeChargeDisputeFundsReinstatedProcessor
```
Comment (sidler-mozilla) [inline subhub/api/webhooks/stripeWebhookPipeline.py]: follow Python file naming conventions already present in the repo: lowercase separated by '_' 1-3 words
this is not java with overly verbose file and class names

File: subhub/api/webhooks/stripeWebhookPipeline.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+
+class StripeWebhookPipeline :
+
+    def __init__(self, payload):
+        assert isinstance(payload, object)
+        self.payload = payload
+
+    def run(self):
+        type = self.payload["type"]
```
Comment (sidler-mozilla) [inline subhub/api/webhooks/stripeWebhookPipeline.py]: type is a keyboard in python that you have obfuscated with this assignment
https://stackoverflow.com/a/2418007

**Comment by jackiemunroe** [comment]
Reviewed code and ran tests locally. Looks good to me +1

File: subhub/api/webhooks/routes/abstract.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+from abc import ABC
+
+import flask
+
+logger = logging.getLogger("abstract_route")
+log_handle = logging.StreamHandler()
+log_handle.setLevel(logging.INFO)
+logformat = logging.Formatter("%(asctime)s - %(name)s - %(levelname)s - %(message)s")
+log_handle.setFormatter(logformat)
+logger.addHandler(log_handle)
```
Comment (bbangert) [inline subhub/api/webhooks/routes/abstract.py]: App-wide logging settings are typically done once near the program setup. Only line 6 is needed per file as a result.

Comment (bbangert) [reply subhub/api/webhooks/routes/abstract.py]: Line 6 in most libraries/apps looks[ like one of these](https://github.com/udhayprakash/PythonMaterial/blob/0df4477e8ce13af1f9b18c1032f8d5f6718b4bec/python2/12_Logging/11_logging.py), I've regularly used method 2 or 1 described there.

File: subhub/api/webhooks/routes/abstract.py [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+    def report_route(self, payload: object, sent_system: str):
+        logger.info(f"data {payload['event_id']} system {sent_system}")
+        existing_event = flask.g.webhook_table.get_event(payload["event_id"])
+        logger.info(f"existing event {existing_event}")
+        if not existing_event:
+            logger.info("no existing event")
+            new_event = flask.g.webhook_table.new_event(
+                event_id=payload["event_id"], sent_system=sent_system
+            )
+            logger.info(f"new event {new_event}")
```
Comment (bbangert) [inline subhub/api/webhooks/routes/abstract.py]: These should probably be debug level statements?

File: subhub/api/webhooks/stripe/controller.py [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+from subhub.api.webhooks.stripe.customer import StripeCustomerCreated
+from subhub.api.webhooks.stripe.charge import StripeChargeSucceededEvent
+from subhub.api.webhooks.stripe.customer import StripeCustomerDeleted
+from subhub.api.webhooks.stripe.customer import StripeCustomerSubscriptionCreated
+from subhub.api.webhooks.stripe.customer import StripeCustomerUpdated
+from subhub.api.webhooks.stripe.subscription import StripeSubscriptionCreated
+from subhub.api.webhooks.stripe.customer import StripeCustomerSubscriptionUpdated
+from subhub.api.webhooks.stripe.customer import StripeCustomerSubscriptionDeleted
+from subhub.api.webhooks.stripe.customer import StripeCustomerSourceExpiring
+from subhub.api.webhooks.stripe.unhandled import StripeUnhandledEvent
```
Comment (bbangert) [inline subhub/api/webhooks/stripe/controller.py]: All imports from the same place should be done together. I'm surprised Black didn't try and aggregate them.

File: subhub/api/webhooks/stripe/controller.py [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+        elif event_type == "customer.deleted":
+            StripeCustomerDeleted(self.payload).run()
+        elif event_type == "customer.source.expiring":
+            StripeCustomerSourceExpiring(self.payload).run()
+        elif event_type == "charge.succeeded":
+            StripeChargeSucceededEvent(self.payload).run()
+        elif event_type == "subscription.created":
+            StripeSubscriptionCreated(self.payload).run()
+        else:
+            StripeUnhandledEvent(self.payload).run()
```
Comment (bbangert) [inline subhub/api/webhooks/stripe/controller.py]: Moving the classes into a dict for dispatch is a fairly common pattern for this type of situation.
```python
event_type_functions = {
   "customer.subscription.created": StripeCustomerSubscriptionCreated,
   "customer.subscription.updated": StripeCustomerSubscriptionUpdated,
  ...
}
```

Then in the run function:
```python
handler = event_type_functions.get(event_type, StripeUnhandledEvent)
handler(self.payload).run()
```

File: subhub/api/webhooks/stripe/customer.py [hunk 10]
```diff
@@ -0,0 +1,149 @@
...
+log_handle = logging.StreamHandler()
+log_handle.setLevel(logging.INFO)
+logformat = logging.Formatter("%(asctime)s - %(name)s - %(levelname)s - %(message)s")
+log_handle.setFormatter(logformat)
+logger.addHandler(log_handle)
+
+
+class StripeCustomerCreated(AbstractStripeWebhookEvent):
+    def run(self):
+        print(f"customer created: {self.payload}")
```
Comment (bbangert) [inline subhub/api/webhooks/stripe/customer.py]: Shouldn't these be logging statements?

File: subhub/subhub_dynamodb.py [hunk 11]
```diff
@@ -69,5 +69,64 @@ def remove_from_db(self, uid: str) -> bool:
...
+            if not sent_system in update_event.sent_system:
+                update_event.sent_system.append(sent_system)
+                update_event.save()
+                return True
+        except DoesNotExist:
+            return False
+
+    def remove_from_db(self, event_id: str) -> bool:
+        try:
+            self.model.get(event_id, consistent_read=True).delete()
```
Comment (bbangert) [inline subhub/subhub_dynamodb.py]: It should be possible to delete without fetching first which will reduce cost and round trips.

File: subhub/tests/unit/stripe/utils.py [hunk 12]
```diff
@@ -0,0 +1,22 @@
+import os
+import json
+
+from subhub.cfg import CFG
+from subhub import secrets
+from subhub.api.webhooks.stripe.pipeline import StripeWebhookEventPipeline
+
+__location__ = os.path.realpath(os.path.dirname(__file__))
```
Comment (bbangert) [inline subhub/tests/unit/stripe/utils.py]: Using dunder scores in this manner is typically reserved for Python or special class overrides, etc.

**Pull request merged**